conversion.go

```diff
@@ -0,0 +1,129 @@
+package safecast
+
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+type Integer interface {
+	~int | ~uint | ~int8 | ~uint8 | ~int16 | ~uint16 | ~int32 | ~uint32 | ~int64 | ~uint64
+}
+
+var ErrOutOfRange = errors.New("out of range")
+
+func ToInt[T Integer](i T) (int, error) {
+	if i > 0 && uint64(i) > math.MaxInt {
+		return 0, fmt.Errorf("%w: %d is greater than math.MaxInt", ErrOutOfRange, i)
```

Comment:
it's creative (short on the writer side) but I wonder if it's backward (to wrap the base error instead of all being the same error type)

Reply:
as I'm wrapping, it can be caught with errors.Is

so I'm unsure what you mean

Do you have an example?

Reply:
yeah https://github.com/fortio/terminal/blob/main/interrupt.go#L30-L32 and how it's used

basically errors.As vs errors.Is but it... works

Reply:
I opened https://github.com/ccoVeille/go-safecast/issues/15 for investigation